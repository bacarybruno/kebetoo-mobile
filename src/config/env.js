/* eslint-disable no-undef */

__DEV__ = false

// TODO: use env variables
export default {
  apiBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.herokuapp.com',
  assetsBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.s3.eu-west-3.amazonaws.com',
}
