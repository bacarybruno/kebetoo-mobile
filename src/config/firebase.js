import { utils } from '@react-native-firebase/app';
import analytics from '@react-native-firebase/analytics';

if (utils().isRunningInTestLab) {
  // do not track analytics for firebase test lab
  // used by google play to run test on publish
  analytics().setAnalyticsCollectionEnabled(false);
}
