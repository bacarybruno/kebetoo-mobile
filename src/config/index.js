export { default as env } from './env';
export { default as strings } from './strings';
