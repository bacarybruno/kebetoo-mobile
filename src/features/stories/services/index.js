export { default as videoEditor } from './editor';
