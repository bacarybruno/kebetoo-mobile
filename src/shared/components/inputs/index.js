export { default as TextInput } from './text'
export { default as PasswordInput } from './password'
export { default as EmojiTextInput } from './emoji'
