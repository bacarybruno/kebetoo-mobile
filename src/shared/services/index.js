import * as http from './http'

export { default as facebookLogin } from './facebook-login'
export { default as googleLogin } from './google-login'
export { http }
