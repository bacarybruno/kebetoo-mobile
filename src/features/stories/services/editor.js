/* eslint-disable max-len */
import { RNFFmpeg, RNFFprobe, RNFFmpegConfig } from 'react-native-ffmpeg';
import RNFetchBlob from 'rn-fetch-blob';

import { getExtension, getFileName } from '@app/shared/helpers/file';
import { env } from '@app/config';

const disableLogs = true;
if (disableLogs) {
  // RNFFmpegConfig.disableStatistics()
  RNFFmpegConfig.disableLogs();
}

// isnpired by https://github.com/shahen94/react-native-video-processing/blob/master/android/src/main/java/com/shahenlibrary/Trimmer/Trimmer.java
const rootDir = RNFetchBlob.fs.dirs.DocumentDir;
const fastPreset = '-preset ultrafast -crf 28 -profile:v baseline';
const pixFmt = '-pix_fmt yuv420p';
const outputDir = `${rootDir}/kebetoo-ffmpeg-cache-dir`;
export const getOutputPath = () => `${outputDir}/${Date.now()}.mp4`;

const videoEditor = {
  executeRawFFmpegCommand: async (command) => {
    const commandArgs = command.split(' ');
    const commandOutput = commandArgs.pop();
    const ffmpegCommand = `-y ${commandArgs.join(' ')} ${commandOutput}`;
    return new Promise((resolve, reject) => (
      RNFFmpeg.executeAsync(ffmpegCommand, (result) => {
        if (result.returnCode !== 0) return reject(result);
        return resolve(result);
      })
    ));
  },

  executeFFmpegCommand: async (input, command, commandOutput) => {
    let output = commandOutput;
    if (input === output) {
      const outputExtension = getExtension(output);
      const outputFileName = getFileName(output).replace(`.${outputExtension}`, '');
      output = `${rootDir}/${outputFileName}-${Date.now()}.${outputExtension}`;
    }
    await videoEditor.executeRawFFmpegCommand(`-i "${input}" ${command} "${output}"`);
    if (input === output) {
      await RNFetchBlob.fs.unlink(input);
      await RNFetchBlob.fs.mv(output, input);
    }
  },

  compose: async (initialParam, ...operations) => {
    let lastResult = initialParam;
    // eslint-disable-next-line no-restricted-syntax
    for (const operation of operations) {
      // eslint-disable-next-line no-await-in-loop
      lastResult = await operation(lastResult);
    }
    return lastResult;
  },

  // cleanup: async (args) => {
  //   await RNFetchBlob.fs.unlink(outputDir).catch(() => { })
  //   await RNFetchBlob.fs.mkdir(outputDir).catch(() => { })
  //   return args
  // },

  // cleanup: async () => {
  //   await RNFetchBlob.fs.unlink(outputDir).catch(() => { })
  //   return videos
  // },

  addSingleVideoEmptySoundIfNeeded: async (video, outputFile = getOutputPath()) => {
    if (!video.mute) return video;
    const generateSilenceArgs = `-f lavfi -i aevalsrc=0 -i ${video.uri} ${fastPreset} -c:v copy -c:a aac -map 0 -map 1:v -shortest ${outputFile}`;
    await videoEditor.executeRawFFmpegCommand(generateSilenceArgs);
    return { ...video, uri: outputFile };
  },

  addMultipleVideosEmptySoundIfNeeded: async (videos) => {
    const results = await Promise.all(videos.map((video) => videoEditor.addSingleVideoEmptySoundIfNeeded(video)));
    return results;
  },

  removeSingleVideoMetadata: async (video, outputFile = getOutputPath()) => {
    const removeMetadataArgs = `-metadata:s:v rotate=0 ${fastPreset}`;
    await videoEditor.executeFFmpegCommand(video.uri, removeMetadataArgs, outputFile);
    return { ...video, uri: outputFile };
  },

  mirrorSingleVideo: async (video, outputFile = getOutputPath()) => {
    if (!video.shouldMirror) return video;
    const mirrorVideoArgs = `-vf hflip ${fastPreset} -c:a copy`;
    await videoEditor.executeFFmpegCommand(video.uri, mirrorVideoArgs, outputFile);
    return { ...video, uri: outputFile };
  },

  flipSingleVideo: async (video) => {
    let result = await videoEditor.removeSingleVideoMetadata(video);
    result = videoEditor.mirrorSingleVideo(result);
    return result;
  },

  flipMultipleVideos: async (videos) => {
    const results = await Promise.all(videos.map((video) => videoEditor.flipSingleVideo(video)));
    return results;
  },

  setSingleVideoSpeed: async (video, outputFile = getOutputPath()) => {
    const { speed, uri: videoUri } = video;
    if (speed === 1) return video;

    let videoSpeedCmd = `[0:v]setpts=${speed.toFixed(1)}*PTS[v]`;
    let audioSpeedCmd = `[0:a]atempo=${speed.toFixed(1)}[a]`;

    if (speed === 0.3) audioSpeedCmd = '[0:a]atempo=0.6,atempo=0.5[a]';
    if (speed < 1) videoSpeedCmd = `[0:v]setpts=${(speed * 10).toFixed(1)}*PTS[v]`;
    else videoSpeedCmd = `[0:v]setpts=${(speed / 10).toFixed(1)}*PTS[v]`;

    const setVideoSpeedArgs = `-filter_complex "${videoSpeedCmd};${audioSpeedCmd}" ${fastPreset} ${pixFmt} -map "[v]" -map "[a]"`;

    await videoEditor.executeFFmpegCommand(videoUri, setVideoSpeedArgs, outputFile);
    return { ...video, uri: outputFile };
  },

  setMultipleVideosSpeed: async (videos) => {
    const results = await Promise.all(videos.map((video) => videoEditor.setSingleVideoSpeed(video)));
    return results;
  },

  mergeMultipleVideos: async (videos, outputFile = getOutputPath()) => {
    if (videos.length === 1) return { uri: videos[0].uri };
    const concatData = videos.map((video) => `file ${video.uri}`).join('\n');
    const concatFilePath = `${outputDir}/concat.txt`;
    await RNFetchBlob.fs.unlink(concatFilePath).catch(() => { });
    await RNFetchBlob.fs.writeFile(concatFilePath, concatData);
    await videoEditor.executeRawFFmpegCommand(`-safe 0 -f concat -i ${concatFilePath} ${fastPreset} -c copy ${outputFile}`);
    return { uri: outputFile };
  },

  compressVideo: async (video, outputFile = getOutputPath()) => {
    const { width: maxWidth, height: maxHeight } = env.stories.aspectRatio;
    const compressVideoArgs = `-filter:v "scale='min(${maxWidth},iw)':min'(${maxHeight},ih)':force_original_aspect_ratio=decrease,pad=${maxWidth}:${maxHeight}:(ow-iw)/2:(oh-ih)/2" ${fastPreset} ${pixFmt}`;
    // const compressVideoArgs = `-vcodec libx264 ${fastPreset} ${pixFmt}`
    // const compressVideoArgs = `-vf "scale=iw/2:ih/2" ${fastPreset} ${pixFmt}`
    await videoEditor.executeFFmpegCommand(video.uri, compressVideoArgs, outputFile);
    return { uri: outputFile };
  },

  boomerang: async (video, outputFile = getOutputPath()) => {
    const boomerangVideoArgs = `-filter_complex "[0:v]reverse,fifo[r];[0:v][0:a][r] [0:a]concat=n=2:v=1:a=1 [v] [a]" ${fastPreset} ${pixFmt} -map "[v]" -map "[a]"`;
    await videoEditor.executeFFmpegCommand(video.uri, boomerangVideoArgs, outputFile);
    return { uri: outputFile };
  },

  reverse: async (video, outputFile = getOutputPath()) => {
    const reverseVideoArgs = `${fastPreset} ${pixFmt} -vf reverse -c:a copy`;
    await videoEditor.executeFFmpegCommand(video.uri, reverseVideoArgs, outputFile);
    return { uri: outputFile };
  },

  slowmo: async (video, videoDuration, outputFile = getOutputPath()) => {
    const bounds = {
      start: 0,
      middleLeft: (videoDuration * 1) / 3,
      middleRight: (videoDuration * 2) / 3,
    };

    const slowmoVideoArgs = ''
      + `-filter_complex "[0:v]trim=${bounds.start}:${bounds.middleLeft},setpts=PTS-STARTPTS[v1];`
      + `[0:v]trim=${bounds.middleLeft}:${bounds.middleRight},setpts=2*(PTS-STARTPTS)[v2];`
      + `[0:v]trim=${bounds.middleRight},setpts=PTS-STARTPTS[v3];`
      + `[0:a]atrim=${bounds.start}:${bounds.middleLeft},asetpts=PTS-STARTPTS[a1];`
      + `[0:a]atrim=${bounds.middleLeft}:${bounds.middleRight},asetpts=PTS-STARTPTS,atempo=0.5[a2];`
      + `[0:a]atrim=${bounds.middleRight},asetpts=PTS-STARTPTS[a3];`
      + `[v1][a1][v2][a2][v3][a3]concat=n=3:v=1:a=1" ${fastPreset} ${pixFmt}`;

    await videoEditor.executeFFmpegCommand(video.uri, slowmoVideoArgs, outputFile);
    return { uri: outputFile };
  },

  getVideoProperties: async (video) => {
    const result = await RNFFprobe.getMediaInformation(video.uri);
    return result.getMediaProperties();
  },

  getDuration: async (video) => {
    const result = await videoEditor.getVideoProperties(video);
    return result.duration;
  },

  isMute: async (video) => {
    const result = await videoEditor.getVideoProperties(video);
    return result.nb_streams === 1;
  },

  overlayVideo: async (overlay, video, outputFile = getOutputPath()) => {
    const overlayVideoArgs = `-i ${overlay} -filter_complex "[0:v][1:v] overlay=0:0:enable='between(t,0,20)'" ${fastPreset} ${pixFmt} -c:a copy`;
    await videoEditor.executeFFmpegCommand(video.uri, overlayVideoArgs, outputFile);
    return { uri: outputFile };
  },

  overlayVideoWithImage: (overlay) => (video, outputFile = getOutputPath()) => videoEditor.overlayVideo(overlay, video, outputFile),
};

export default videoEditor;
