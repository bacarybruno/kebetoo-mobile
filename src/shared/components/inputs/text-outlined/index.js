import React from 'react'

import { TextInput, Typography } from '@app/shared/components'
import { metrics } from '@app/theme'
import { useAppStyles } from '@app/shared/hooks'

import createThemedStyles from './styles'

export const TextOutlinedInput = ({
  autoFocus = false,
  onChange,
  text,
  label,
  editable = true,
  borderless = true,
  maxNumberOfLines = 1,
  placeholder,
  maxLength = 180,
  inputRef,
  textStyle,
  wrapperStyle,
  error,
  fieldName,
}) => {
  const styles = useAppStyles(createThemedStyles)
  return (
    <>
      <Typography
        style={styles.label}
        type={Typography.types.headline6}
        text={label}
      />
      <TextInput
        multiline
        autoFocus={autoFocus}
        placeholder={placeholder}
        onValueChange={onChange}
        fieldName={fieldName}
        borderless={true}
        error={error}
        returnKeyType="default"
        textStyle={[styles.textInput, textStyle]}
        wrapperStyle={[styles.textInputWrapper, !borderless && styles.border, !borderless && error && styles.error, wrapperStyle]}
        inputWrapperStyle={styles.inputWrapper}
        maxLength={maxLength}
        editable={editable}
        defaultValue={text}
        ref={inputRef}
        numberOfLines={Math.min(
          Math.floor(metrics.screenHeight / 75),
          maxNumberOfLines,
        )}
      />
    </>
  )
}

export default React.memo(TextOutlinedInput)
