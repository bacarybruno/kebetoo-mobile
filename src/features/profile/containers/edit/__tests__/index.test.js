import { act } from 'react-test-renderer'

import setupTest from '@app/config/jest-setup'
import { strings } from '@app/config'
import { TextOutlinedInput } from '@app/shared/components/inputs/text-outlined'
import { api } from '@app/shared/services'

import EditProfile, { fieldNames } from '../index'

beforeEach(jest.clearAllMocks)

const givenEditProfile = setupTest(EditProfile)({
  route: {
    params: {
      field: 'username',
    },
  },
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
})

it('renders EditProfile', () => {
  const { wrapper } = givenEditProfile()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('updates profile', async () => {
  const { wrapper, props } = givenEditProfile()
  const [displayName, username, _, bio] = wrapper.root.findAllByType(TextOutlinedInput)

  await act(async () => {
    await displayName.props.onChange('jest', fieldNames.fullName)
    await username.props.onChange('@jestuser', fieldNames.username)
    await bio.props.onChange('Hello from the test side!', fieldNames.bio)
    await wrapper.root.findByProps({ text: strings.general.save.toUpperCase() }).props.onPress()
  })

  expect(api.authors.update).toBeCalledTimes(1)
  expect(api.authors.update).toBeCalledWith(expect.any(String), {
    username: '@jestuser',
    displayName: 'jest',
    bio: 'Hello from the test side!',
  })
  expect(props.navigation.goBack).toBeCalledTimes(1)
})

it('aborts profile update if username exists', async () => {
  api.authors.getByUsername.mockResolvedValue([{ id: 'fake-id-123' }])
  const { wrapper, props } = givenEditProfile()
  const [displayName, username, _, bio] = wrapper.root.findAllByType(TextOutlinedInput)

  expect(username.props.error).toBeNull()

  await act(async () => {
    await displayName.props.onChange('jest', fieldNames.fullName)
    await username.props.onChange('@jestuser', fieldNames.username)
    await bio.props.onChange('Hello from the test side!', fieldNames.bio)
    await wrapper.root.findByProps({ text: strings.general.save.toUpperCase() }).props.onPress()
  })

  expect(api.authors.update).not.toBeCalled()
  expect(props.navigation.goBack).not.toBeCalled()
  expect(username.props.error).not.toBeNull()
})
