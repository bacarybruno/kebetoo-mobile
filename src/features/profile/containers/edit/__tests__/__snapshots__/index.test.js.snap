// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditProfile 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 255)",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginLeft": 16,
                    "width": 24,
                  },
                  Object {
                    "marginLeft": 0,
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Object {
                    "color": "rgba(0,0,0,0.87)",
                    "fontFamily": "icomoon",
                    "fontSize": 20,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
                text="kebeticon-chevron-left"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 22,
                      "fontWeight": "400",
                      "letterSpacing": 0.34375,
                      "lineHeight": 28,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      Array [
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
              >
                Edit profile
              </Text>
              <ActivityIndicator
                animating={false}
                color="rgba(0, 122, 255, 255)"
                hidesWhenStopped={true}
                size="small"
                style={
                  Object {
                    "marginLeft": 8,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "rgba(0, 122, 255, 255)",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 36,
              "justifyContent": "center",
              "minWidth": 64,
              "paddingHorizontal": 16,
            },
            undefined,
            false,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#FFFFFF",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": -0.32,
                "lineHeight": 21,
              },
              Array [
                Object {
                  "color": "rgba(0, 122, 255, 255)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "600",
                },
                undefined,
              ],
            ]
          }
        >
          SAVE
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 24,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginTop": 16,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 16,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 30,
                  "width": 30,
                },
                Object {
                  "height": 115,
                  "width": 115,
                },
                undefined,
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://avatars1.githubusercontent.com/u/14147533",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 122, 255, 255)",
                  "borderRadius": 1000,
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
              testID="image-avatar-https://avatars1.githubusercontent.com/u/14147533"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 0, 0, 0.4)",
                  "borderRadius": 100,
                  "height": "100%",
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 255)",
                    "fontFamily": "Ionicons",
                    "fontSize": 35,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
                text="ionicon-camera"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              false,
              false,
              false,
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.54)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "marginLeft": 0,
                },
              ],
            ]
          }
        >
          Full Name
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 255)",
              "borderBottomColor": "rgba(229, 229, 234, 255)",
              "borderBottomWidth": 1,
              "borderColor": "transparent",
              "borderRadius": 0,
              "borderWidth": 1,
              "minHeight": 48,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 16,
                },
                Object {
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoFocus={false}
              blurOnSubmit={false}
              defaultValue="Bruno Bodian"
              editable={true}
              maxLength={60}
              multiline={true}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Bruno Bodian"
              placeholderTextColor="rgba(174, 174, 178, 255)"
              rejectResponderTermination={true}
              returnKeyType="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "flex": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "height": "100%",
                    "letterSpacing": -0.32,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "color": "rgba(0,0,0,0.87)",
                      "paddingBottom": 8,
                    },
                    undefined,
                  ],
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={null}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              false,
              false,
              false,
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.54)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "marginLeft": 0,
                },
              ],
            ]
          }
        >
          Username
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 255)",
              "borderBottomColor": "rgba(229, 229, 234, 255)",
              "borderBottomWidth": 1,
              "borderColor": "transparent",
              "borderRadius": 0,
              "borderWidth": 1,
              "minHeight": 48,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 16,
                },
                Object {
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoFocus={false}
              blurOnSubmit={false}
              editable={true}
              maxLength={24}
              multiline={true}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Username"
              placeholderTextColor="rgba(174, 174, 178, 255)"
              rejectResponderTermination={true}
              returnKeyType="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "flex": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "height": "100%",
                    "letterSpacing": -0.32,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "color": "rgba(0,0,0,0.87)",
                      "paddingBottom": 8,
                    },
                    undefined,
                  ],
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={null}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              false,
              false,
              false,
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.54)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "marginLeft": 0,
                },
              ],
            ]
          }
        >
          Email
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 255)",
              "borderBottomColor": "rgba(229, 229, 234, 255)",
              "borderBottomWidth": 1,
              "borderColor": "transparent",
              "borderRadius": 0,
              "borderWidth": 1,
              "minHeight": 48,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 16,
                },
                Object {
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoFocus={false}
              blurOnSubmit={false}
              defaultValue="bacarybruno@kebetoo.com"
              editable={false}
              maxLength={254}
              multiline={true}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Email"
              placeholderTextColor="rgba(174, 174, 178, 255)"
              rejectResponderTermination={true}
              returnKeyType="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "flex": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "height": "100%",
                    "letterSpacing": -0.32,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "color": "rgba(0,0,0,0.87)",
                      "paddingBottom": 8,
                    },
                    undefined,
                  ],
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={null}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              false,
              false,
              false,
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.54)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "marginLeft": 0,
                },
              ],
            ]
          }
        >
          Bio
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 255)",
              "borderBottomColor": "rgba(229, 229, 234, 255)",
              "borderBottomWidth": 1,
              "borderColor": "transparent",
              "borderRadius": 0,
              "borderWidth": 1,
              "minHeight": 48,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingLeft": 16,
                },
                Object {
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoFocus={false}
              blurOnSubmit={false}
              editable={true}
              maxLength={160}
              multiline={true}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Share your interests or hobbies"
              placeholderTextColor="rgba(174, 174, 178, 255)"
              rejectResponderTermination={true}
              returnKeyType="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "flex": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "height": "100%",
                    "letterSpacing": -0.32,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "color": "rgba(0,0,0,0.87)",
                      "paddingBottom": 8,
                    },
                    undefined,
                  ],
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={null}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
